roles/example-cnf-app/tasks/trex/app.yaml

```diff
@@ -1,105 +1,116 @@
 ---
-- name: set local fact for trex pod networks
+- name: Set local fact for trex pod networks
   set_fact:
     networks_trex: []
     packet_gen_net: "{{ packet_generator_networks if enable_lb|bool else cnf_app_networks }}"
     trex_app_run_passed: false
+
 # TODO(skramaja): This logic to be improved for multiple
 # networks, lets fail it if the length is > 1
 - fail:
     msg: "Need to rewrite the mac merging logic"
   when: "packet_gen_net|length != 1"
-- name: create network list for trex with hardcoded macs
+
+- name: Create network list for trex with hardcoded macs
   set_fact:
     networks_trex: "{{ networks_trex + [ item | combine({ 'mac': trex_mac_list[idx:idx+item.count] }) ] }}"
   loop: "{{ packet_gen_net }}"
   loop_control:
     index_var: idx
 
-- name: create cr for trex server
+- name: Create CR for trex server
   k8s:
     definition: "{{ lookup('template', 'trex-server-cr.yaml.j2') }}"
-- name: check trex pod status to be running
+
+- name: Check TRex pod status to be running
   k8s_info:
     namespace: "{{ cnf_namespace }}"
     kind: Pod
     label_selectors:
       - example-cnf-type=pkt-gen
   register: trex_result
+  until: trex_result | json_query('resources[*].status.phase') | unique == ["Running"]
   retries: 60
   delay: 5
-  until:
-    - "trex_result.resources|length > 0"
-    - "'status' in trex_result.resources[0]"
-    - "'phase' in trex_result.resources[0].status"
-    - "trex_result.resources[0].status.phase == 'Running'"
-- name: set trex run status if passed
+
+- name: Set TRex run status if the previous step passed
   set_fact:
     trex_app_run_passed: true
-  when: trex_result.resources|length > 0
+  when: trex_result.resources | length > 0
 
-- name: trexapp block
-  when: enable_trex_app|bool
+- name: TRexApp block
   block:
-  - name: set trex app cr name
-    set_fact:
-      duration: 120
-  - name: create cr for trex app
-    k8s:
-      definition: "{{ lookup('template', 'trex-app-cr.yaml.j2') }}"
-    vars:
-      packet_rate: 10kpps
-      packet_size: 64
+    - name: Set TRex app CR name
+      set_fact:
+        duration: 120
+
+    - name: Create CR for TRex app
+      k8s:
+        definition: "{{ lookup('template', 'trex-app-cr.yaml.j2') }}"
+      vars:
+        packet_rate: 10kpps
+        packet_size: 64
+
+    - name: Wait for TRex app CR to be created
+      shell: |
+        {{ oc_tool_path }} get trexapp.examplecnf.openshift.io -o yaml -n {{ cnf_namespace }} {{ trex_app_cr_name }}
+      register: trex_app_cr_installation
+      retries: 10
```

Comment:
Is the number of retries enough? Is there any difference between adding this wait of max 200s and increasing the retries of the next check on events?

Reply:
@capolrik these are two different tasks: to wait for the application to be up and to wait for the first traffic to be running. I have no idea if waiting for 200s is enough, we never checked it before :) I added the task to wait for CR from scratch here.